rootfs-builder/ubuntu/config.sh

```diff
@@ -0,0 +1,17 @@
+#
+# Copyright (c) 2017 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+
+# architecture to build the rootfs for
+ARCH=${ARCH:-"amd64"}
+
+# url to download rootfs from
+ARCHIVE_URL=${ARCHIVE_URL:-"http://archive.ubuntu.com/ubuntu/"}
+
+# this should be ubuntu's codename eg Xenial for 16.04
+OS_NAME=${OS_NAME:-"xenial"}
```

Comment:
This isn't really a comment specific to this PR - just a reminder to all...

I appreciate this is following the existing convention, but we do all need to think about how we'll manage these versions and code names as one day, they will become invalid.

See:

- https://github.com/clearcontainers/runtime/wiki/Handling-new-and-end-of-life-distro-versions
- https://github.com/clearcontainers/tests/issues/857

For `osbuilder`, we could create a special build target which called a shell script and checked `ARCHIVE_URL` + `OS_NAME` maybe (and there are bound to be distro-specific bits to add to that). But we could then atleast call this build rule as part of the CI.

Reply:
I agree we should have a mechanism to detect EOL distro version.....However I won't be able to work on that issue before 17th(have my exams till then) I can hopefully work on it after that, if the issue is not already resolved by then.

Reply:
Hi @ydjainopensource - thanks. I added this comment as your PR just reminded me of the general issue. Please don't feel you have to work on that feature unless it's something that interests you ;)

I suspect there will need to be a bit of discussion about how best to solve this problem for all the distros we handle but it is something that has "caught us out" before so it would be good to find a general and robust solution to the problem.

Reply:
I loved love to work on it. Do you have any ideas as to how we should go about tackling them? I have worked only on apt based distros.